internal/cluster/finalizing_stages.go

```diff
@@ -0,0 +1,81 @@
+package cluster
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+)
+
+const (
+	generalWaitTimeout = 70 * time.Minute
+	longWaitTimeout    = 10 * time.Hour
+	shortWaitTimeout   = 10 * time.Minute
+)
+
+var finalizingStagesTimeoutsDefaults = map[models.FinalizingStage]time.Duration{
+	models.FinalizingStageWaitingForFinalizing:                    longWaitTimeout,
+	models.FinalizingStageWaitingForClusterOperators:              longWaitTimeout,
+	models.FinalizingStageAddingRouterCa:                          generalWaitTimeout,
+	models.FinalizingStageWaitingForOlmOperators:                  shortWaitTimeout,
+	models.FinalizingStageApplyingManifests:                       shortWaitTimeout,
+	models.FinalizingStageWaitingForOlmOperatorsCsv:               generalWaitTimeout,
+	models.FinalizingStageWaitingForOlmOperatorsCsvInitialization: generalWaitTimeout,
+	models.FinalizingStageDone:                                    shortWaitTimeout,
+}
+
+var finalizingStages = []models.FinalizingStage{
+	models.FinalizingStageWaitingForFinalizing,
+	models.FinalizingStageWaitingForClusterOperators,
+	models.FinalizingStageAddingRouterCa,
+	models.FinalizingStageWaitingForOlmOperators,
+	models.FinalizingStageApplyingManifests,
+	models.FinalizingStageWaitingForOlmOperatorsCsvInitialization,
+	models.FinalizingStageWaitingForOlmOperatorsCsv,
+	models.FinalizingStageDone,
+}
+
+func convertStageToEnvVar(stage models.FinalizingStage) string {
+	return fmt.Sprintf("FINALIZING_STAGE_%s_TIMEOUT", strings.ReplaceAll(strings.ToUpper(string(stage)), " ", "_"))
```

Comment:
Do we have env vars for all the stage timeouts? If not this seems like a bit of overkill.

Reply:
Yes we do.